apps/frontend/src/app/Formula/optimization.ts

```diff
@@ -357,7 +378,7 @@ export function constantFold(
 
           if (numericValue !== f([]))
             // Skip vacuous values
-            formulaOperands.push(constant(numericValue))
+            formulaOperands.unshift(constant(numericValue))
```

Comment:
If it will be reordered anyway, we shouldn't need `unshift`, which is normally O(n).